### PR TITLE
Small improvements v2?

### PR DESCRIPTION
**Pull request by cubasepp**
```
irb(main):002:0> Orchparty.upgrade(cluster_name: 'staging-shared-3', application_name: 'metoda-account', force_variable_definition: nil, file_name: file)
Executing command: helm upgrade --debug --namespace metoda-account --kube-context staging-shared-3 metoda-account /tmp/d20220712-7211-1cy7jmu
       user     system      total        real
   0.001625   0.000059   1.743343 ( 12.865122)
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /home/michael/work/metoda-infra/me/kube/config
WARNING: Kubernetes configuration file is world-readable. This is insecure. Location: /home/michael/work/metoda-infra/me/kube/config
upgrade.go:142: [debug] preparing upgrade for metoda-account
upgrade.go:150: [debug] performing update for metoda-account
upgrade.go:322: [debug] creating upgraded release for metoda-account
client.go:218: [debug] checking 32 resources for changes
client.go:510: [debug] Patch PodDisruptionBudget "metoda-account-api-console" in namespace metoda-account
client.go:510: [debug] Patch PodDisruptionBudget "metoda-account-api" in namespace metoda-account
client.go:510: [debug] Patch PodDisruptionBudget "metoda-account-redis" in namespace metoda-account
client.go:510: [debug] Patch PodDisruptionBudget "metoda-account-ui-v1-console" in namespace metoda-account
client.go:510: [debug] Patch PodDisruptionBudget "metoda-account-ui-v1" in namespace metoda-account
client.go:510: [debug] Patch PodDisruptionBudget "metoda-account-ui-v2-console" in namespace metoda-account
client.go:510: [debug] Patch PodDisruptionBudget "metoda-account-ui-v2" in namespace metoda-account
client.go:501: [debug] Looks like there are no changes for Service "metoda-account-api-console"
client.go:501: [debug] Looks like there are no changes for Service "metodaccount-api"
client.go:501: [debug] Looks like there are no changes for Service "metoda-account-redis"
client.go:501: [debug] Looks like there are no changes for Service "metoda-account-ui-v1-console"
client.go:501: [debug] Looks like there are no changes for Service "metoda-account-ui-v1"
client.go:501: [debug] Looks like there are no changes for Service "metoda-account-ui-v2-console"
client.go:501: [debug] Looks like there are no changes for Service "metoda-account-ui-v2"
client.go:510: [debug] Patch Deployment "metoda-account-api-console" in namespace metoda-account
client.go:510: [debug] Patch Deployment "metoda-account-api" in namespace metoda-account
client.go:510: [debug] Patch Deployment "metoda-account-delayed-job" in namespace metoda-account
client.go:510: [debug] Patch Deployment "metoda-account-redis" in namespace metoda-account
client.go:501: [debug] Looks like there are no changes for Deployment "metoda-account-ui-v1-console"
client.go:501: [debug] Looks like there are no changes for Deployment "metoda-account-ui-v1"
client.go:510: [debug] Patch Deployment "metoda-account-ui-v2-console" in namespace metoda-account
client.go:510: [debug] Patch Deployment "metoda-account-ui-v2" in namespace metoda-account
client.go:510: [debug] Patch CronJob "metoda-account-cron-db-consistency" in namespace metoda-account
client.go:510: [debug] Patch CronJob "metoda-account-cron-generate-permissions-files" in namespace metoda-account
client.go:510: [debug] Patch CronJob "metoda-account-cron-generate-product-config-report" in namespace metoda-account
client.go:510: [debug] Patch CronJob "metoda-account-cron-remove-product-configurations" in namespace metoda-account
client.go:510: [debug] Patch Ingress "metoda-account-api-console" in namespace metoda-account
client.go:510: [debug] Patch Ingress "metoda-account-api" in namespace metoda-account
client.go:510: [debug] Patch Ingress "metoda-account-ui-v1-console" in namespace metoda-account
client.go:510: [debug] Patch Ingress "metoda-account-ui-v1" in namespace metoda-account
client.go:510: [debug] Patch Ingress "metoda-account-ui-v2-console" in namespace metoda-account
client.go:510: [debug] Patch Ingress "metoda-account-ui-v2" in namespace metoda-account
upgrade.go:157: [debug] updating status for upgraded release for metoda-account

```

Problem here is the helm command. 
Maybe our service is to big, not sure :shrug: 
Maybe the connection to the cluster is the problem.

I wonder about the patches in ingress/cronjob and so on any ideas?

In the end I guess I could fetch maybe 0.5-1 second more? :man_shrugging:

### CODE REVIEW

File: lib/orchparty/kubernetes_application.rb [hunk 1]
```diff
@@ -188,14 +188,17 @@ def get_binding(params)
       end
 
       def build_chart(chart)
-        params = chart._services.map {|s| app_config.services[s.to_sym] }.map{|s| [s.name, s]}.to_h
+        params = chart._services.each.with_object({}) do |service, params|
```
Comment (cubasepp) [inline lib/orchparty/kubernetes_application.rb:191]: previous map does an map and an map to build an hash.

File: lib/orchparty/kubernetes_application.rb [hunk 2]
```diff
@@ -322,25 +325,13 @@ def print(method)
     end
   end
 
-  def combine_charts(app_config)
```
Comment (cubasepp) [inline lib/orchparty/kubernetes_application.rb:325]: From my test this function makes from

```
[metoda-account-api-console
...
metoda-account]
```

this:

```[metoda-account]```


with my solution I have the same result. 
In the end we should build Charts...

**Comment by timonugmann** [comment]
> I wonder about the patches in ingress/cronjob and so on any ideas?

Not sure about the ingress, but for cronjob we made it explicitly to be update-able every time, because previously we had some issues with the other approach (more specifically, previously it was so that if you deploy a cronjob with an erroring image, then you had to manually patch it).

I guess to have a better estimate of how to speed up helm we'd need to know how much time each such request takes.

It doesn't really look to me that we have a huge-huge service (although maybe being able to break it down would be good); It could also be simply that helm is slow somehow.

Or maybe it really is so that we should check the diff of charts ourselves before we run helm, and only run on those that changed: that would speed things up

**Comment by cubasepp** [comment]
Okay. Will close this PR and move https://youtrack.metoda.com/issue/INFRA-1677

New ticket required. Check Helm :man_shrugging:

**Pull request closed without merging**